### PR TITLE
Minor fixes to help evaluate ELMo port.

### PR DESCRIPTION
**Pull request by brendan-ai2**
- Display loss when running evaluate.
  - Fixes https://github.com/allenai/allennlp/issues/1241
- Log when sentences are dropped due to exceeding the maximum length.

### CODE REVIEW

**Comment by brendan-ai2** [comment]
A tiny one. Please take a look!

File: allennlp/commands/evaluate.py [hunk 1]
```diff
@@ -100,10 +100,27 @@ def evaluate(model: Model,
...
+                has_loss = "loss" in output_dict
+            num_batches += 1
+
             metrics = model.get_metrics()
+
+            if has_loss:
+                loss = output_dict["loss"].item()
+                metrics["loss"] = loss
+                cumulative_loss += loss
+
```
Comment (joelgrus) [inline allennlp/commands/evaluate.py]: is the consistency a real worry? I think you could make this a tiny bit simpler, something like

```
loss_count = 0
total_loss = 0.0

# ...

loss = model(**batch).get("loss")

# ...

if loss is not None:
    metrics["loss"] = loss.item()
    total_loss += loss.item()
    loss_count += 1

# ...

if loss_count > 0:
    final_metrics["loss"] = total_loss / loss_count
```

I feel like that's cleaner, what do you think?

**Comment by brendan-ai2** [comment]
Thanks!
